src/Libraries/RevitNodes/Elements/Viewport.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Autodesk.DesignScript.Geometry;
+using Autodesk.Revit.DB;
+using Revit.Elements.Views;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+
+namespace Revit.Elements
+{
+    public class Viewport : Element
+    {
+        #region Internal properties
+
+        /// <summary>
+        /// An internal reference to the Viewport
+        /// </summary>
+        internal Autodesk.Revit.DB.Viewport InternalViewport
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalViewport; }
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Private constructor for the Element
+        /// </summary>
+        /// <param name="viewport"></param>
+        private Viewport(Autodesk.Revit.DB.Viewport viewport)
+        {
+            SafeInit(() => InitViewport(viewport));
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Initialize a Viewport element
+        /// </summary>
+        /// <param name="viewport"></param>
+        private void InitViewport(Autodesk.Revit.DB.Viewport viewport)
+        {
+            InternalSetViewport(viewport);
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the Viewport property, element id, and unique id
+        /// </summary>
+        /// <param name="viewport"></param>
+        private void InternalSetViewport(Autodesk.Revit.DB.Viewport viewport)
+        {
+            this.InternalViewport = viewport;
+            this.InternalElementId = viewport.Id;
+            this.InternalUniqueId = viewport.UniqueId;
+        }
+
+        #endregion
+
+        #region Public properties
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Creates a new Viewport at a given location on a sheet.
+        /// </summary>
+        /// <param name="sheet">The Sheet on which the new Viewport will be placed.</param>
+        /// <param name="view">The view shown in the Viewport.</param>
+        /// <param name="location">The new Viewport will be centered on this point.</param>
+        /// <returns>The new Viewport.</returns>
+        public static Viewport Create(Sheet sheet, Revit.Elements.Views.View view, Autodesk.DesignScript.Geometry.Point location)
```

Comment:
name is not Dynamo-ish, could probs be `Viewport.BySheetViewLocation` or `BySheetView`